daniel-matthew/scripts/article.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var articles = [];
+
+function Article (rawDataObj) {
+  // TODONE: Use the JS object passed in to complete this constructor function:
+  // Save ALL the properties of `rawDataObj` into `this`
+  this.authorName = rawDataObj.author;
+  this.authorUrl = rawDataObj.authorUrl;
+  this.articleTitle = rawDataObj.title;
+  this.articleBody = rawDataObj.body;
+  this.publishedOn = rawDataObj.publishedOn;
+}
+
+Article.prototype.toHtml = function() {
+  var $newArticle = $('article.template').clone();
+  $newArticle.removeClass('template');
+  /* TODO: This cloned article still has a class of template.
+  However, in our modules.css stylesheet, we gave all elements
+  with a class of template a display of none. Let's make
+  sure we're not accidentally hiding our cloned article! */
+  if (!this.publishedOn) $newArticle.addClass('draft');
+  $newArticle.data('category', this.category);
+
+  /* TODO: Now use jQuery traversal and setter methods to fill in the rest
+  of the current template clone with properties from this particular Article instance.
+  We need to fill in:
+    1. author name,
+    2. author url,
+    3. article title,
+    4. article body, and
+    5. publication date. */
+  $newArticle.find('h1').html(this.articleTitle);
```

Comment:
Since the title is plain text with no HTML tags, you should probably use `.text()` instead of `.html()`.